presidio-anonymizer/presidio_anonymizer/engine/engine.py

```diff
@@ -0,0 +1,24 @@
+class Engine:
+    # TODO this needs to be implemented currently a stab.
+    # TODO replace analyze_results with domain results
+    # Notice the document Omri created, it impacts the implementation
+    def __init__(
+            self,
+            text: str,
+            # TODO change to domain object
+            transformations: list[str],
+            # TODO change to domain object
+            analyze_results: list[str],
+    ):
+        self.transformations = transformations
+        self.analyze_results = analyze_results
+        self.text = text
+        self._end_point = len(text)
+
+    def run(self):
```

Comment:
if we conform with how the analyzer looks like, then this module should be called `AnonymizerEngine` and the main method `anonymize`.